localstack/services/redshift/provider.py

```diff
@@ -21,15 +23,11 @@ def fix_keys(o, **kwargs):
                             v[k[:-1]] = v.pop("item")
             return o
 
-        result = itemize_orig(data, *args, **kwargs)
+        result = fn(data, *args, **kwargs)
         recurse_object(result, fix_keys)
         return result
 
-    itemize_orig = redshift_responses.itemize
-    redshift_responses.itemize = itemize
 
-
-def start_redshift(port=None, asynchronous=False):
-    port = port or config.service_port("redshift")
-    apply_patches()
-    return start_moto_server("redshift", port, name="Redshift", asynchronous=asynchronous)
+class RedshiftProvider(RedshiftApi, ServiceLifecycleHook):
+    def on_after_init(self):
+        apply_patches()
```

Comment:
it seems to me we could just remove apply_patches all together, and define the `@patch` decorated methods on module level. lazy-init will make sure the patches are applied only when the service is loaded.

Reply:
Good catch, updated. 👍 

I guess over time we'll see whether we want to introduce a patch apply/unapply in the service lifecycle, but for now there doesn't seem to be a strong use case. Also makes the implementation here a bit simpler, as we don't need to rely on the `ServiceLifecycleHook`.

Reply:
the `@patch` actually adds the `Patch` object to the function, so you could do something like
```python
def toggle_patches():
    itemize.patch.undo()
```

another way can be to use the `Patches` collection to aggregate the used patches. `patches` could become a conventional object as part of the module

```python
patches = Patches([itemize.patch, ...])
```
then you can call `module.patches.apply()` or `module.patches.undo()`

lots of flexibility :-)